### DISCUSSION TITLE
Multiple UseFetch from within Composable.

### DISCUSSION THREAD

**Discussion by XPAULUSX** [nuxt/nuxt#16754] [Questions]
Hi, 

I'm trying to call UseFetch twice from within a Composable (two different endpoints). I need to be able to retrieve data from two separate API endpoints (I'm using the server directory to handle this and pass off the requests to the relevant APIs) however, as soon as there are two instances of UseFetch, I get the server error "Nuxt instance not available", works absolutely perfect with one.

Is UseFetch by design only to be used once for each page/component? Similar to "asyncData" from Nuxt 2?

I have a workaround where I can just call both APIs from within my server route but would rather do it this way if possible?

Any help greatly appreciated as usual, thank you.

**Comment by sevillaarvin** [comment]
I don't think so, at least it's not mentioned in the docs. Could you provide a reproduction repo?

**Comment by XPAULUSX** [comment]
Sure thing!

https://github.com/XPAULUSX/UseFetchRepro

**Reply by sevillaarvin** [reply]
I couldn't figure it out. Saw this instead:

https://github.com/nuxt/framework/discussions/1851
https://github.com/nuxt/framework/issues/1852

> This is by design. `useFetch` shouldn't be triggered in a method; like most composables, it is designed to be run in `setup()`. If you need an on-demand fetch use `$fetch()` instead. Alternatively, you can implement like so:
> 
> ```js
> function useMyThing () {
>   const { data, refresh } = 
>   const [{ data: organization, refresh: refreshOrganization }, { data: repos, refresh: refreshRepos }] = Promise.all([
>     useFetch('https://api.github.com/orgs/nuxt'),
>     useFetch('https://api.github.com/orgs/nuxt/repos'),
>   ])
>   return {
>     organization,
>     repos,
>     fetch: () => Promise.all([refreshOrganization(), refreshRepos()])
>   }
> }
> ```

**Reply by XPAULUSX** [reply]
Thanks for that, I couldn't figure it out either, it looks as if both of my composables are being called from within script setup tag so I was a bit lost with the issue!

Using $fetch seems to work for me for now anyhow, hopefully can leave this open and see if anyone else can help out.

Thank you for your time.
